### PR TITLE
Fix: isRequired validation of checkbox widget

### LINKED ISSUE #991: [Bug] The isRequired field doesn't work in checkbox

### PR DESCRIPTION
**Pull request by jsartisan**
## Description
- The form submit button was not getting disabled even when checkbox's isRequired toggle was on.
- Disabled submit button was getting clicked.

Fixes #991 

## Type of change
- Bug

## How Has This Been Tested?
- Manually verified
- Added cypress test.

### CODE REVIEW

File: app/client/src/widgets/CheckboxWidget.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ class CheckboxWidget extends BaseWidget<CheckboxWidgetProps, WidgetState> {
   static getDerivedPropertiesMap(): DerivedPropertiesMap {
     return {
       value: `{{this.isChecked}}`,
+      isValid: `{{ this.isRequired ? !!this.isChecked : true }}`,
```
Comment (hetunandu) [inline app/client/src/widgets/CheckboxWidget.tsx:43]: Can we show on the Checkbox widget that this input is required similar to input widget?

Comment (jsartisan) [reply app/client/src/widgets/CheckboxWidget.tsx:43]: Screenshot 2020-11-25 at 1 01 27 PM

Styled the error state like this.

**Comment by mohanarpit** [comment]
@jsartisan Can you please improve the title of this PR? I'm asking because this is what will go into the changelog and we want it to be:
1. Concise. It should not be more than a single line. 
2. Explanatory of the bug that was fixed.

**Comment by jsartisan** [comment]
@mohanarpit Updated the title.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**